lib/libzfs/libzfs_sendrecv.c

```diff
@@ -298,7 +298,7 @@ send_iterate_snap(zfs_handle_t *zhp, void *arg)
 	const char *from = sd->fromsnap;
 	const char *to = sd->tosnap;
 
-	assert(snapname != (NULL + 1));
+	assert(snapname - 1 != NULL);
```

Comment:
If `snapname - 1 == NULL`, Undefined Behavior has been already invoked, and nasal daemons will take over your system daemons :~)

The actual fix should be changing the line above:
```
char *snapname = strrchr(zhp->zfs_name, '@') + 1;
```
which is causing UB.  You should break that line, and check for NULL _before_ adding 1:

```
char *snapname;

...
snapname = strrchr(zhp->zfs_name, '@');
ASSERT(snapname != NULL)
snapname++;
```

Reply:
Good point. Fixed.